File: heart/src/main/java/com/vantagecircle/heartrate/activity/ActivityModule.java
package com.vantagecircle.heartrate.activity;

import android.app.Activity;
import android.content.Context;

import com.vantagecircle.heartrate.annotation.ActivityContext;

import dagger.Module;
import dagger.Provides;

/**
 * Created by bapidas on 14/11/17.
 */
@Module
public class ActivityModule {
    private Activity mActivity;

    public ActivityModule(Activity mActivity) {
        this.mActivity = mActivity;
    }

    @Provides
    @ActivityContext
    Context provideContext() {
        return mActivity;
    }

    @Provides
    Activity provideActivity() {
        return mActivity;
    }
}

File: app/src/main/java/com/bapidas/heartrate/di/module/ActivityModule.kt
package com.bapidas.heartrate.di.module

import android.app.Activity
import android.content.Context
import com.bapidas.heartrate.di.annotation.ActivityContext
import dagger.Module
import dagger.Provides

/**
 * Created by bapidas on 14/11/17.
 */
@Module
class ActivityModule(private val mActivity: Activity) {
    @Provides
    @ActivityContext
    fun provideContext(): Context {
        return mActivity
    }

    @Provides
    fun provideActivity(): Activity {
        return mActivity
    }
}